analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Branching.LearnConstraint.cs

```diff
@@ -513,18 +537,27 @@ public void Branching_LearnsObjectConstraint_ConstantPattern_False(string expres
         }
 
         [DataTestMethod]
-        [DataRow("arg is 42", "int")]
         [DataRow("arg is 42", "int?")]
         [DataRow("arg is 42", "T")]
-        [DataRow("arg is 42", "TStruct")]
         public void Branching_LearnsObjectConstraint_ConstantPattern_ValueTypes_InputIsNotReferenceType(string expression, string argType)
         {
             var validator = CreateIfElseEndValidatorCS(expression, OperationKind.ConstantPattern, argType);
-            validator.ValidateTag("If", x => x.AllConstraints.Should().ContainSingle().Which.Should().Be(ObjectConstraint.NotNull));
-            validator.ValidateTag("Else", x => x.Should().BeNull());
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.Should().HaveNoConstraints());
             validator.TagValues("End").Should().BeEquivalentTo(new SymbolicValue[] { SymbolicValue.NotNull, null });
         }
 
+        [DataTestMethod]
+        [DataRow("arg is 42", "int")]
+        [DataRow("arg is 42", "TStruct")]
+        public void Branching_LearnsObjectConstraint_ConstantPattern_ValueTypes_InputNonNullableValueType(string expression, string argType)
+        {
+            var validator = CreateIfElseEndValidatorCS(expression, OperationKind.ConstantPattern, argType);
+            validator.ValidateTag("If", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+            validator.ValidateTag("Else", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
+            validator.TagValues("End").Should().BeEquivalentTo(new[] { SymbolicValue.NotNull });
```

Comment:
Using `ValidateTag` implies that there's only single item. It throws otherwise.
```suggestion
            validator.ValidateTag("Else", x => x.Should().HaveOnlyConstraint(ObjectConstraint.NotNull));
```